### PR TITLE
Update github.com/go-openapi/jsonreference to drop github.com/PuerkitoBio/purell

### PR DESCRIPTION
**Pull request by dims**
Nice win. Only change in jsonreference is to drop this library, not used anywhere else! 

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: vendor/github.com/go-openapi/jsonreference/internal/normalize_url.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+func lowercaseHost(u *url.URL) {
+	if len(u.Host) > 0 {
+		u.Host = strings.ToLower(u.Host)
+	}
+}
+
+func removeDefaultPort(u *url.URL) {
+	if len(u.Host) > 0 {
+		scheme := strings.ToLower(u.Scheme)
+		u.Host = rxPort.ReplaceAllStringFunc(u.Host, func(val string) string {
```
Comment (liggitt) [inline vendor/github.com/go-openapi/jsonreference/internal/normalize_url.go:50]: I'm not 100% sure this port regex would do the right thing on an ipv6 host containing colons, but I verified this is the same as the existing PuerkitoBio normalization code

**Comment by liggitt** [comment]
```diff
 {
-	"directDependencies": 219,
+	"directDependencies": 217,
 	"transitiveDependencies": 258,
 	"totalDependencies": 311,
 	"maxDepthOfDependencies": 30
```

/lgtm
/approve

File: vendor/github.com/go-openapi/jsonreference/internal/normalize_url.go (hunk 1 above)
Comment (dims) [reply vendor/github.com/go-openapi/jsonreference/internal/normalize_url.go:50]: cc @aojea what say you!? :)

We can take a fix to the `go-openapi` repo if we know this is a bug and/or we have a fix.

**Pull request merged**